llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7699,6 +7695,10 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
   if (Instruction *New = foldSignBitTest(I))
     return New;
 
+  if (auto *PN = dyn_cast<PHINode>(Op0))
```

Comment:
There is no guarantee that the phi node will be op0 if the other operand is not constant, so we should handle both. Can you please add a commuted test?